services/datamanager/datacapture/data_capture_queue.go

```diff
@@ -0,0 +1,121 @@
+package datacapture
+
+import (
+	"sync"
+
+	"github.com/pkg/errors"
+	v1 "go.viam.com/api/app/datasync/v1"
+)
+
+var (
+	// ErrQueueClosed indicates that a Push or Pop was attempted on a closed queue.
+	ErrQueueClosed = errors.New("queue is closed")
+	// MaxFileSize is the maximum size in bytes of a data capture file.
+	MaxFileSize = int64(65536)
+)
+
+// Queue is a persistent queue of SensorData backed by a series of datacapture.Files.
+type Queue struct {
+	Directory string
+	MetaData  *v1.DataCaptureMetadata
+	nextFile  *File
+	lock      *sync.Mutex
+	files     []*File
+	closed    bool
+}
+
+// NewQueue returns a new Queue.
+func NewQueue(dir string, md *v1.DataCaptureMetadata) *Queue {
+	return &Queue{
+		Directory: dir,
+		files:     []*File{},
+		lock:      &sync.Mutex{},
+		MetaData:  md,
+	}
+}
+
+// Push pushes item onto q.
+func (q *Queue) Push(item *v1.SensorData) error {
+	q.lock.Lock()
+	defer q.lock.Unlock()
+
+	if q.closed {
+		return ErrQueueClosed
+	}
+
+	if q.nextFile == nil {
+		nextFile, err := NewFile(q.Directory, q.MetaData)
+		if err != nil {
+			return err
+		}
+		q.nextFile = nextFile
+	} else if q.nextFile.Size() > MaxFileSize || item.GetBinary() != nil {
```

Comment:
[q] I'm wondering why this shouldn't be < ?

Reply:
Jk, it seems like you're skipping this file if it's greater than the maxSize? In that case, why are we also skipping binary readings?

Reply:
It's > because we want to move onto a new file if the file is > our max size